File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
    int n = s.size();   
    if(n % 2 != 0){
        return false;
    } else {
        for(int i = 0; i < n; i++){
            if(s[i] == '('){
                st.push(')');
            } else if(s[i] == '{'){
                st.push('}');
            } else if(s[i] == '['){
                st.push(']');
            } else if (s[i] == ')' || s[i] == ']' || s[i] == '}'){
                if(st.empty()){
                    return false;
                } else {
                    char top = st.top();
                    if(top == s[i]){
                        st.pop();
                    } else {
                        return false;
                    }
                }
            }
        }
    }

    if(st.empty()){
        return true;
    } else {
        return false;
    }
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        stack = []

        for i in range(n):
            days = 0
            while stack and temperatures[i] > temperatures[stack[-1]]:
                old_idx = stack.pop()
                days = i - old_idx
                res[old_idx] = days
            
            stack.append(i)
        

        return res
            